From 5b7c4ed099d8273c7cde874de49677e623421578 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 19 Nov 2012 16:23:06 +0200
Subject: [PORT FROM MAIN] Telephony: Add support for RIL_UNSOL_OEM_HOOK_RAW in CommandsInterface

BZ: 22242

The Android framework doesn't fully support the unsolicited notification
RIL_UNSOL_OEM_HOOK_RAW.

Change-Id: I065757555d5cd02f2fe4d4bcc6c5986cf5d7a48b
Orig-Change-Id: I74ff6d7ade79bf0dcd45a6f71c803cbd3ed5c477
Signed-off-by: Frederic Predon <frederic.predon@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/75448
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
Reviewed-by: Lucas, Guillaume <guillaume.lucas@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../internal/telephony/CommandsInterface.java      |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/CommandsInterface.java b/src/java/com/android/internal/telephony/CommandsInterface.java
index 27b0d68..3981441 100644
--- a/src/java/com/android/internal/telephony/CommandsInterface.java
+++ b/src/java/com/android/internal/telephony/CommandsInterface.java
@@ -417,6 +417,17 @@ public interface CommandsInterface {
     void registerForSignalInfo(Handler h, int what, Object obj);
     void unregisterForSignalInfo(Handler h);
 
+     /**
+     * Sets the handler for OEM Hook Raw Notifications.
+     * Unlike the register* methods, there's only one notification handler
+     *
+     * @param h Handler for notification message.
+     * @param what User-defined message code.
+     * @param obj User object.
+     */
+     void setOnUnsolOemHookRaw(Handler h, int what, Object obj);
+     void unSetOnUnsolOemHookRaw(Handler h);
+
     /**
      * Registers the handler for CDMA number information record
      * Unlike the register* methods, there's only one notification handler
-- 
1.7.4.1

